cfdm/data/subarray/mixin/biquadraticlatitudelongitudeinterpolation.py

```diff
@@ -0,0 +1,242 @@
+import numpy as np
+
+from .quadraticgeographicinterpolation import QuadraticGeographicInterpolation
+
+
+class BiQuadraticLatitudeLongitudeInterpolation(
+    QuadraticGeographicInterpolation
+):
+    """Mixin class for bi-quadratic latitude longitude interpolation.
+
+    See CF appendix J "Coordinate Interpolation Methods".
+
+    .. versionadded:: (cfdm) 1.9.TODO.0
+
+    """
+
+    def _bi_quadratic_latitude_longitude_interpolation(
+        self,
+        lat_a,
+        lon_a,
+        lat_b,
+        lon_b,
+        lat_c,
+        lon_c,
+        lat_d,
+        lon_d,
+        ce1,
+        ca1,
+        ce2,
+        ca2,
+        ce3,
+        ca3,
+        location_use_3d_cartesian,
+        d2,
+        d1,
+    ):
+        """Bi-quadratic interpolation of geographic coordinates.
+
+        A two-dimensional quadratic method for interpolation of the
+        geographic coordinates latitude and longitude, typically used
+        for remote sensing products with geographic coordinates on the
+        reference ellipsoid.
+
+        See CF appendix J "Coordinate Interpolation Methods".
+
+        .. versionadded:: (cfdm) 1.9.TODO.0
+
+        :Parameters:
+
+            lat_a, lon_a: `numpy.ndarray`
+                The latitude and longitude of the first tie point in
+                index space of subsampled dimension 1 (in the sense of
+                CF appendix J Figure J.1).
+
+            lat_b, lon_b: `numpy.ndarray`
+                The latitude and longitude of the second tie point in
+                index space of subsampled dimension 2 (in the sense of
```

Comment:
```suggestion
                index space of subsampled dimension 1 (in the sense of
```

I think? Else there are two keywords with the same description in the docstring here, so something is probably not right?

Reply:
Great spot. I think all of these dimension numbers need checking in many places. I'll get on to it ...

Reply:
(... and they *really* matter - if you get the dimensions switched the code can still work, giving plausible-but-wrong answers - the worst possible outcome.)

Reply:
How about: https://github.com/NCAS-CMS/cfdm/pull/168/commits/5637964c31b01d6674b621f05898c9b76f3e1971

Reply:
Ah yes, I like the descriptions from that new commit much better as the variable names correspond directly to them, whereas before the variable naming seemed arbitrary, at least without consulting the relevant figure. Nice!

Unfortunately it seems you have left in the final parenthesis to each description, so those need removing still, as per the one 'suggestion' I have made to illustrate in-line. Otherwise this can be be marked as resolved.

Reply:
Thanks - typos fixed.